### ISSUE TITLE
NovaLXD hook failed: "config-changed" and next steps after installation

### ISSUE THREAD

**Issue by indispeq** [conjure-up/conjure-up#361]
Thank you for a great piece of software. I found out about it ironically at the NYC AWS summit after talking to one of the Canonical engineers. I installed it on an intel NUC core i5, 16GB RAM, 120GB SSD. I did `sudo lxd init` as well as configured dnsmasq to be able to resolve the `.lxd` domain in order to be able to ping containers.

My NIC is on `eno1` and I have `lxdbr0`. I left all settings at default so the neutrong-gateway seems to have set itself up to `eth1`. 

The installation finishes successfully , and I pressed 'submit' on all tasks with keypairs and glance-images.

Now I have the following at the bottom of `juju status`:

```
UNIT                     WORKLOAD  AGENT      MACHINE  PORTS           PUBLIC-ADDRESS  MESSAGE
ceph-mon/0               active    idle       0                        10.219.41.150   Unit is ready and clustered
ceph-mon/1               active    idle       1                        10.219.41.198   Unit is ready and clustered
ceph-mon/2               active    idle       2                        10.219.41.152   Unit is ready and clustered
ceph-osd/0               active    idle       3                        10.219.41.230   Unit is ready (1 OSD)
ceph-osd/1               active    idle       4                        10.219.41.52    Unit is ready (1 OSD)
ceph-osd/2               active    idle       5                        10.219.41.157   Unit is ready (1 OSD)
ceph-radosgw/0           active    executing  6        80/tcp          10.219.41.109   (update-status) Unit is ready
glance/0                 active    idle       7        9292/tcp        10.219.41.59    Unit is ready
keystone/0               active    idle       8        5000/tcp        10.219.41.190   Unit is ready
mysql/0                  active    idle       9                        10.219.41.241   Unit is ready
neutron-api/0            active    idle       10       9696/tcp        10.219.41.70    Unit is ready
neutron-gateway/0        active    idle       11                       10.219.41.237   Unit is ready
nova-cloud-controller/0  active    idle       12       8774/tcp        10.219.41.120   Unit is ready
nova-compute/0           active    idle       13                       10.219.41.129   Unit is ready
  lxd/0                  error     idle                                10.219.41.129   hook failed: "config-changed"
  neutron-openvswitch/0  active    idle                                10.219.41.129   Unit is ready
openstack-dashboard/0    active    idle       14       80/tcp,443/tcp  10.219.41.50    Unit is ready
rabbitmq-server/0        active    idle       15       5672/tcp        10.219.41.203   Unit is ready
```

Is this LXD error of hook failed a problem?

Also, how do I actually reach the horizon dashboard or get into any of the containers to launch instances? I tried curling to the container openstac-dashboard, but I get this:
```
user@xenial-1:~$ curl 10.219.41.50
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<html>
 <head>
  <title>Index of /</title>
 </head>
 <body>
<h1>Index of /</h1>
  <table>
   <tr><th valign="top"><img src="/icons/blank.gif" alt="[ICO]"></th><th><a href="?C=N;O=D">Name</a></th><th><a href="?C=M;O=A">Last modified</a></th><th><a href="?C=S;O=A">Size</a></th><th><a href="?C=D;O=A">Description</a></th></tr>
   <tr><th colspan="5"><hr></th></tr>
<tr><td valign="top"><img src="/icons/folder.gif" alt="[DIR]"></td><td><a href="html/">html/</a></td><td align="right">2016-08-20 21:55  </td><td align="right">  - </td><td>&nbsp;</td></tr>
   <tr><th colspan="5"><hr></th></tr>
</table>
<address>Apache/2.4.18 (Ubuntu) Server at 10.219.41.50 Port 70</address>
</body></html>
```

What are next steps to be able to test this openstack deployment on a single NUC?

Thanks for the help!

**Comment by battlemidget** [comment]
For horizon the url should be:

```
curl 10.219.41.50/horizon
```

What versions of juju, lxd, and conjure-up are installed?

**Comment by indispeq** [comment]
Thank you for that. I managed to access the dashboard by installing nginx on the host and doing a `proxy_pass` to the container's IP.

I noticed that I had to issue a `sysctl -w fs.inotify.max_user_instances=8192` because of all the open containers otherwise I could not restart/reload nginx. Maybe this is something that should be considered by conjure-up otherwise there may be problems managing the host (just a sidenote but I think it could be helpful).

I got to the dashboard so I will now try out networking and creating an ext-net of my own. I assume in this all-in-one setup the best thing to do is flat networking and give my external router's IP as gateway?

For the hook failed error, here are the version:
juju: 2.0-beta12-xenial-amd64
lxd: 2.0.3
conjure-up:  conjure-up 2.0.0

Ubuntu 16.04, no ZFS for LXD (I only have one 120GB SSD).

**Comment by battlemidget** [comment]
A OpenStack install with NovaLXD should create it's own bridge that your OpenStack will utilize so there is no need to do configure any additional networking. Everything should just work at that point.

I'll make a note of the max_user_instances that may needing to be increased in the docs.

**Comment by indispeq** [comment]
Thank you, making progress. Very impressive to be able to run all this just in containers!

I was able to create an LXD based VM , but unfortunately I cannot create an external network for the floating IPs on my actual physical network. I did manage to ssh into the instance using the pre-defined 
ext-net. How would I be able to integrate the setup with my existing physical network? (flat networking to create a new external network fails). Or is this not a legitimate use case?

The pre-defined ubuntu-net is not getting the right DNS servers, it gets 127.0.0.1 by default which does not allow resolution correctly. I modified the DNS servers in the ubuntu-subnet by default to be 8.8.8.8 and 8.8.4.4 and my instance can run an apt-get update. Not sure if 127.0.0.1 should be able to resolve DNS queries. If so, is there something missing on my end for this?
